### DISCUSSION TITLE
Does Normalize plugin work with Windows?

### DISCUSSION THREAD

**Discussion by wigofran** [gpodder/gpodder#1546] [Q&A]
I tried using normalize audio plugin downloading official binary (normalize.nongnu.org/).
I noticed that python source calls different binaries than Windows "normalize.exe", so I put the binary in PATH and I create batchs files (normalize-mp3.bat, normalize-ogg.bat, normalize-audio.bat).
In command line, I can use normalize.exe and the batches that I created. However, when I try to use normalize option in menu, I got error:

[gpodder.log] ERROR: Uncaught exception: Traceback (most recent call last):
  File "gPodder\data\share\gpodder\extensions\normalize_audio.py", line 113, in convert_episodes
    self._convert_episode(episode)
  File "gPodder\data\share\gpodder\extensions\normalize_audio.py", line 96, in _convert_episode
    p = util.Popen(cmd, cwd=episode.channel.save_dir)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "gPodder\data\lib\python3.11\site-packages\gpodder\util.py", line 2168, in __init__
    super(Popen, self).__init__(*args, **kwargs)  # Python 2 syntax
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:/msys64/home/appveyor/_gpodder_build_root/mingw32/lib/python3.11/subprocess.py", line 1026, in __init__
  File "C:/msys64/home/appveyor/_gpodder_build_root/mingw32/lib/python3.11/subprocess.py", line 1538, in _execute_child
FileNotFoundError: [WinError 2]

**Comment by auouymous** [comment]
It has Windows specific code so it should work. The extension found those three bat files, otherwise it would have disabled itself. It searches for exe or bat, but I'm wondering if python's Popen also appends those extensions, or if this code has never worked on Windows. Can you try running `python -c 'import subprocess; subprocess.Popen(["cmd"])'`, without the `.exe`, and see if it opens a command prompt?

**Reply by wigofran** [reply]
Thanks for replying.
It doesn't open command but modifying quotes it works.

![Screenshot 2023-08-25 08-04-08 clink-git](https://github.com/gpodder/gpodder/assets/16245018/ba6aa1f0-4bc5-43c5-8431-66bf1b6fe1a8)

**Comment by auouymous** [comment]
Okay, so Popen does append ".exe". See if it appends ".bat" if you replace "cmd" with "normalize-ogg"?

**Comment by wigofran** [comment]
Popen includes "exe" so on Windows it won't work.

![Screenshot 2023-08-28 07-17-45 clink-git](https://github.com/gpodder/gpodder/assets/16245018/9930d81e-d433-4b8f-9ed3-1f7e898d2f90)


The official Windows distribution of Normalize does not have the 'normalize-ogg.exe' binary. So I'll duplicate the "normalize.exe" file and rename it to test.

**Comment by wigofran** [comment]
It worked. I copied and renamed the exe file. Thanks!

**Comment by auouymous** [comment]
https://gpodder.github.io/docs/extensions/normalizeaudio.html now mentions this.
